### ISSUE TITLE
(PROD116) - (ALL) - Annual Report Queries Not correct?

### ISSUE THREAD

**Issue by altrdgenetics** [CurrentlyNotSet/SERB#497]
The new Annual Report document stats are not all correct. Please see the differences between SMDS 2.0 and SMDS 3.0. I provided two different date ranges for the reports to compare.

(NOT INCLUDED... 12 pages of printouts)

So glad these were dumped on us and not told WHICH cases were missing... Anthony is looking through it to see the problems, sounds like we will bill for this no matter the outcome since it was dumped on us.

**Comment by altrdgenetics** [comment]
Justin Has just sent this.....

> XLN Team,
> 
> I added the following text.
> 
> Since you’re having XLN “tweak” the Annual Report – the total number of ULP dismissals should include: ULP, ERC and JWD cases.  Right now – the ERC and JWD are not accounted for when they are dismissed by the Board.
> 
> Thank you,
> Justin Brown


Looks like we will def be billing for this.

**Comment by anthonyperk** [comment]
Changes to Annual Report
---------------------------------------------------
One - Cases Filed - Total Cases
-------------------------------
This is a count of the total cases.  since ULP, ERC, and JWD were off, the total cases is skewed.  This will be fixed in the release


Six - Case Filed - ULP
-----------------------
Query looks fine, look at the code

SELECT COUNT(*) FROM ULPCase WHERE (ULPCase.fileDate BETWEEN + startDate + AND + endDate +) AND (ULPCase.caseType = 'ULP')




Nine - Agency Activites - State Mediator Appointed
---------------------------------------------------
Change in the query (MEDCase.fileDate -> MEDCase.mediatorAppointedDate)

SELECT COUNT(*) FROM MEDCase WHERE (MEDCase.mediatorAppointedDate BETWEEN + startDate + AND + endDate +) AND (MEDCase.stateMediatorAppointedID IS NOT NULL) AND (MEDCase.mediatorAppointedDate IS NOT NULL)



Ten - Agency Activites - Federal Mediator Appointed
---------------------------------------------------
Old DB had 2 BLANK casenumbers which accounted for the 2 extra in the old count.  The new DB is correct

OneThree - Agency Activities - Strikes
---------------------------------------
Old DB had 1 BLANK casenumber which accounted for the 1 extra in the old count.  The new DB is correct


OneFour - Agency Activites - Elections Held
---------------------------------------------------
Change to the WHERE statement (MEDCase.pollingStartDate -> MEDCase.pollingEndDate)

SELECT COUNT(*) FROM REPCase WHERE (REPCase.pollingEndDate BETWEEN + startDate + AND + endDate +)



OneEight - Agency Activites - Board opinions issued
---------------------------------------------------
The old DB had 2 cases with Opinions:
2015-REP-01-0004 (opinion 2015-001)
2015-ULP-02-0042 (opinion 2015-002)



TwoSeven - Final Dispositions - Voluntary recognition requests certified
-------------------------------------------------------------------------
Change to the SELECT statement (SELECT Count(*) -> SELECT COUNT(DISTINCT (BoardMeeting.caseYear + '-' + BoardMeeting.caseType + '-' + BoardMeeting.caseMonth + '-' + BoardMeeting.caseNumber)))

SELECT COUNT(DISTINCT (BoardMeeting.caseYear + '-' + BoardMeeting.caseType + '-' + BoardMeeting.caseMonth + '-' + BoardMeeting.caseNumber)) FROM BoardMeeting LEFT JOIN REPCase ON (BoardMeeting.caseYear = REPCase.caseYear AND BoardMeeting.caseType = REPCase.caseType AND BoardMeeting.caseMonth = REPCase.caseMonth AND BoardMeeting.caseNumber = REPCase.caseNumber) WHERE (BoardMeeting.boardMeetingDate BETWEEN + startDate + AND + endDate +) AND (REPCase.boardActionType LIKE '%cert vr%' OR REPCase.boardActionType LIKE '%cert opt-in vr%')



TwoNine - Final Dispositions - Misc. Representation activites
-------------------------------------------------------------
Change to the SELECT statement (SELECT Count(*) -> SELECT COUNT(DISTINCT (BoardMeeting.caseYear + '-' + BoardMeeting.caseType + '-' + BoardMeeting.caseMonth + '-' + BoardMeeting.caseNumber)))

SELECT COUNT(DISTINCT (BoardMeeting.caseYear + '-' + BoardMeeting.caseType + '-' + BoardMeeting.caseMonth + '-' + BoardMeeting.caseNumber)) FROM BoardMeeting LEFT JOIN REPCase ON (BoardMeeting.caseYear = REPCase.caseYear AND BoardMeeting.caseType = REPCase.caseType AND BoardMeeting.caseMonth = REPCase.caseMonth AND BoardMeeting.caseNumber = REPCase.caseNumber) WHERE (BoardMeeting.boardMeetingDate BETWEEN + startDate + AND + endDate +) AND (REPCase.boardActionType LIKE '%approve%' OR REPCase.boardActionType LIKE '%appr ac%' OR REPCase.boardActionType LIKE '%appr jtac%' OR REPCase.boardActionType LIKE '%dism ac%' OR REPCase.boardActionType LIKE '%dism jtac%' OR REPCase.boardActionType LIKE '%appr uc%' OR REPCase.boardActionType LIKE '%dism uc%' OR REPCase.boardActionType LIKE '%revoke%')



ThreeOne - Final Dispositions - ULP Charges Dismissed
-----------------------------------------------------
Change to the WHERE statement (ULPCase.recommendation LIKE 'DM%' -> BoardMeeting.recommendation LIKE 'DM%')

SELECT COUNT(*) FROM BoardMeeting LEFT JOIN ULPCase ON (BoardMeeting.caseYear = ULPCase.caseYear AND BoardMeeting.caseType = ULPCase.caseType AND BoardMeeting.caseMonth = ULPCase.caseMonth AND BoardMeeting.caseNumber = ULPCase.caseNumber) WHERE ((BoardMeeting.boardMeetingDate BETWEEN + startDate + AND + endDate +) AND BoardMeeting.recommendation LIKE 'DM%') AND BoardMeeting.caseType = 'ULP' AND ULPCase.dismissalDate IS NOT NULL)



ThreeThree - Final Dispositions - ULP Charges deffered/jurisdiction retained 
----------------------------------------------------------------------------
Change to the WHERE statement (ULPCase.recommendation LIKE 'DF%' -> BoardMeeting.recommendation LIKE 'DF%')

SELECT COUNT(*) FROM BoardMeeting LEFT JOIN ULPCase ON (BoardMeeting.caseYear = ULPCase.caseYear AND BoardMeeting.caseType = ULPCase.caseType AND BoardMeeting.caseMonth = ULPCase.caseMonth AND BoardMeeting.caseNumber = ULPCase.caseNumber) WHERE (BoardMeeting.recommendation LIKE 'DF%' AND BoardMeeting.boardMeetingDate BETWEEN + startDate + AND + endDate +) AND (ULPCase.caseType != 'ERC' OR ULPCase.caseType != 'JWD')



ThreeNine - Filings and Appointments - Mediator Appointments
-------------------------------------------------------------
Old DB had 2 BLANK casenumbers which accounted for the two extra in the old count.  The new DB is correct



FourTwo - FY YEAR Notices to Negotiate - Statutory - Initial
------------------------------------------------------------
Old DB had 2 BLANK casenumbers which accounted for the two extra in the old count.  The new DB is correct

FourFour - FY YEAR Notices to Negotiate - Statutory - Successor
------------------------------------------------------------
Two case numbers in the old DB have had their negotiation type changed
2016-MED-11-1335 (changed from SE to ZE by Mary on 04/25/2017)
2016-MED-11-1336 (changed from SE to ZE by Mary on 04/25/2017)

FourFive - FY YEAR Notices to Negotiate - Statutory - Total
------------------------------------------------------------
Since FourTwo is off by 2 because of the blank case numbers, this will also be off by two


FourEight - FY YEAR Notices to Negotiate - MADs - Successor
------------------------------------------------------------
Old DB had 1 BLANK casenumber which accounted for the 1 extra in the old count.  The new DB is correct

FourFive - FY YEAR Notices to Negotiate - MADs - Total
------------------------------------------------------------
Since FourEight is off by 1 because of the blank case number, this will also be off by 1

FiveZero - FY YEAR Notices to Negotiate - Total - Initial
------------------------------------------------------------
Since FourTwo is off by 2 because of the blank case numbers, this will also be off by 2

FiveTwo - FY YEAR Notices to Negotiate - Total - Successor
------------------------------------------------------------
Since FourEight is off by 1 because of the blank case number, this will also be off by 1

FiveThree - FY YEAR Notices to Negotiate - Total - Initial
------------------------------------------------------------
Since FourFive is off by 2 because of the blank case numbers, and FourNine is off by 1 because of the blank case number, this will also be off by 3


SevenThree and SevenFour - Results of Fact Finding - Acceptances, Rejections
-----------------------------------------------------------------------------
These will be swapped in code


SevenFive - Fact Finding Cases by Employer Type - Police
--------------------------------------------------------
Change in WHERE statement (MEDCase.FFEmployerType -> MEDCase.FFEmployeeType)

SELECT COUNT(*) FROM MEDCase WHERE (MEDCase.FFReportIssueDate BETWEEN + startDate + AND + endDate +) AND (MEDCase.FFEmployeeType = 'Police')


SevenSix - Fact Finding Cases by Employer Type - Fire
--------------------------------------------------------
Change in WHERE statement (MEDCase.FFEmployerType -> MEDCase.FFEmployeeType)

SELECT COUNT(*) FROM MEDCase WHERE (MEDCase.FFReportIssueDate BETWEEN + startDate + AND + endDate +) AND (MEDCase.FFEmployeeType = 'Fire')


SevenSeven - Fact Finding Cases by Employer Type - Teaching
-----------------------------------------------------------
Change in WHERE statement (MEDCase.FFEmployerType -> MEDCase.FFEmployeeType)

SELECT COUNT(*) FROM MEDCase WHERE (MEDCase.FFReportIssueDate BETWEEN + startDate + AND + endDate +) AND (MEDCase.FFEmployeeType = 'Teaching')


SevenEight - Fact Finding Cases by Employer Type - Nursing
----------------------------------------------------------
Change in WHERE statement (MEDCase.FFEmployerType -> MEDCase.FFEmployeeType)

SELECT COUNT(*) FROM MEDCase WHERE (MEDCase.FFReportIssueDate BETWEEN + startDate + AND + endDate +) AND (MEDCase.FFEmployeeType = 'Nursing')

SevenNine - Fact Finding Cases by Employer Type - Other
--------------------------------------------------------
Change in WHERE statement (MEDCase.FFEmployerType -> MEDCase.FFEmployeeType)

SELECT COUNT(*) FROM MEDCase WHERE (MEDCase.FFReportIssueDate BETWEEN + startDate + AND + endDate +) AND (MEDCase.FFEmployeeType = 'Other')


EightSix - Representation Summary of Activity - Elections Held
--------------------------------------------------------------
Ran both old and new queries and got the same list (count of 21)


EightSeven - Representation Summary of Activity - Unit Determination elections held
-----------------------------------------------------------------------------------
Change to the WHERE Statement (REPCase.electionType3 IS NOT NULL -> REPCase.ProfessionalNonProfessional)

SELECT COUNT(*) FROM REPCase WHERE (REPCase.pollingEndDate BETWEEN + startDate + AND + endDate +) AND (REPCase.ProfessionalNonProfessional = 1)


EightEight - Representation Summary of Activity - Choices for Representation
----------------------------------------------------------------------------
Issues with the following case numbers:
2015-REP-06-0063 - Missing Polling Start Date in 3.0 (should be 09/22/2015)
2015-REP-11-0109 - Was a Prof/NonProf in 2.0, now is a standard election (election info changed by Angela on 3/29/2017)
2016-REP-01-0002 - Missing from count due to who prevailed as 'No Representitive', which the query doesnt count (election info changed by Angela on 3/29/2017)
2016-REP-01-0011 - Missing from count due to who prevailed as 'No Representitive', which the query doesnt count (election info changed by Angela on 3/29/2017)

Change to WHERE statement

SELECT COUNT(*) FROM REPCase WHERE (REPCase.pollingEndDate BETWEEN  + startDate + AND + endDate +) AND (REPCase.pollingStartDate IS NOT NULL) AND (REPCase.pollingEndDate IS NOT NULL) AND (REPCase.resultWhoPrevailed != 0 AND REPCase.resultWhoPrevailed IS NOT NULL) 
 

EightNine - Representation Summary of Activity - Approximate number of eligible voters
--------------------------------------------------------------------------------------
Since there have been election result changes in previous queries listed above, there is a good chance the tallys were changed/updated as well


NineZero - Representation Summary of Activity - Voter Turnout
--------------------------------------------------------------------------------------
Since there have been election result changes in previous queries listed above, there is a good chance the tallys were changed/updated as well


NineOne - Representation Summary of Activity - Certification via Voluntary Request
--------------------------------------------------------------------------------------
Change to the SELECT statement (SELECT Count(*) -> COUNT(DISTINCT (BoardMeeting.caseYear + '-' + BoardMeeting.caseType + '-' + BoardMeeting.caseMonth + '-' + BoardMeeting.caseNumber)))

SELECT COUNT(DISTINCT (BoardMeeting.caseYear + '-' + BoardMeeting.caseType + '-' + BoardMeeting.caseMonth + '-' + BoardMeeting.caseNumber)) FROM BoardMeeting LEFT JOIN REPCase ON (BoardMeeting.caseYear = REPCase.caseYear AND BoardMeeting.caseType = REPCase.caseType AND BoardMeeting.caseMonth = REPCase.caseMonth AND BoardMeeting.caseNumber = REPCase.caseNumber) WHERE (BoardMeeting.boardMeetingDate BETWEEN + startDate + AND + endDate +) AND (REPCase.boardActionType LIKE '%cert vr%' OR REPCase.boardActionType LIKE '%cert opt-in vr%')  


NineThree - Unfair Labor Practice Summary of Activity - ULP Charges Dismissed
------------------------------------------------------------------------------
Change in WHERE statement (ULPCase.recommendation -> BoardMeeting.recommendation)

SELECT COUNT(*) FROM BoardMeeting LEFT JOIN ULPCase ON (BoardMeeting.caseYear = ULPCase.caseYear AND BoardMeeting.caseType = ULPCase.caseType AND BoardMeeting.caseMonth = ULPCase.caseMonth AND BoardMeeting.caseNumber = ULPCase.caseNumber) WHERE (BoardMeeting.boardMeetingDate BETWEEN + startDate + AND + endDate +) AND (BoardMeeting.recommendation LIKE 'DM%') AND (BoardMeeting.caseType = 'ULP') AND (ULPCase.dismissalDate IS NOT NULL)



NineFive - Unfair Labor Practice Summary of Activity - Probable Cause Findings
-------------------------------------------------------------------------------
When I ran the queries again, they both resulted in the same count of 28


NineSix - Unfair Labor Practice Summary of Activity - Deferrals to Arbitration
------------------------------------------------------------------------------
Change in WHERE and FROM statements

SELECT COUNT(*) FROM BoardMeeting WHERE (BoardMeeting.boardMeetingDate BETWEEN + startDate + AND + endDate +) AND (BoardMeeting.recommendation LIKE 'DF%') AND (BoardMeeting.caseType = 'ULP')


OneZeroThree - Hearings Section Summary of Activity - ALJPOs/ALJRDs
--------------------------------------------------------------------
This is due to the case 2015-REP-01-0004 is missing the Proposed Rec Issued date in 3.0 (which was 08/17/2015, but the activity in 3.0 shows that the case information was changed in late 2015)



OneZeroFive - Hearings Section Summary of Activity - Hearings Held
-------------------------------------------------------------------
The following cases dont have a hearing scheduled in 3.0 (but do in 2.0):

2015-REP-02-0014
2015-REP-03-0036
2015-REP-05-0056
2016-REP-02-0016
